docs/content/direct/common-setup-hub-chart.md

```diff
@@ -0,0 +1,80 @@
+# Common Setup - hub chart
+
+For more details please see [here](./hub-chart.md).
+
+This documents explains how to use KubeStellar Hub chart to deploy a new instance of KubeStellar
+with a choice of user-defined KubeFlex Control Planes (CPs).
+
+The information provided is specific for the following release:
+
+```shell
+export KUBESTELLAR_VERSION=0.23.0-alpha.2
+```
+
+## Create a Kind cluster
+
+The setup of KubStellar via the Hub chart requires the existance of at least one cluster
+to be used for the deployment of the chart.
+
+For convenience, a new local Kind cluster that satisfies the requirements for KubeStellar setup
+and that can be used to exercises the [examples](./examples.md) can be created with the following command:
+
+```shell
+sudo su <<EOF
+bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${KUBESTELLAR_VERSION}/scripts/create-kind-cluster-with-SSL-passthrough.sh) --name kubeflex --port 9443
+EOF
+```
+
+## Install KubeStellar Hub chart
+
+A KubeStellar Hub installation compatible with the common setup suitable for Common Setup described in the [examples](https://docs.kubestellar.io/release-0.22.0/direct/examples/) could be achieved with the following command:
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set-json='ITSes=[{"name":"its1"}]' \
+  --set-json='WDSes=[{"name":"wds1"}]'
+```
+
+Remember to include the option `--set "kubeflex-operator.isOpenShift=true"`, when deploying in an OpenShift cluster.
+
+## Obtain the kubeconfigs to access the KubeFlex Control Planes

```

Comment:
Dang, this section is long and fiddly. The addition of this complexity is required because this PR replaces the use of `kflex create` with creating ControlPlane API objects but `kflex create` did more than that (it also modified the current kubeconfig). We switched from `kflex create` with PCH use to Helm chart instantiation, in part because Helm charts are better regarding parameters (i.e., a chart declares all its parameters in a well-defined place and provides default value, while a PCH does neither of those things). We use `kflex` or a published Helm chart in order to avoid requiring the user to fetch KubeStellar scripts. If the user were fetching KubeStellar scripts then the complexity in this section could be nicely wrapped up in a script with a simple interface. And the deficiencies of PCHes wrt parameters would be less important because the user would use scripts instead of directlly using PCHes.

Reply:
I originally thought that using PCH doesn't really gives you major advantage if you are going to automate your deployments through scripts in any case. If you use scripts you can go go ahead and install the exact thing PCH is doing but directly with more control over the process .

Reply:
There are different "you"s. I distinguish between the creators/maintainers (contributors, if you like) of KubeStellar and the _users_ of KubeStellar. It is unfriendly to make the _users_ of KubeStellar write and maintain scripts that have a lot of detailed knowledge of implementation details that us KubeStellar creators/maintainers are trying to wrap up and hide. I wish that we could provide scripts for users to invoke.